recipes/libcurl/all/conanfile.py

```diff
@@ -0,0 +1,414 @@
+               "with_libpsl": [True, False],
+               "with_largemaxwritesize": [True, False],
+               "with_nghttp2": [True, False],
+               "with_brotli": [True, False]}
+    default_options = {'shared': False,
+                       'fPIC': True,
+                       'with_openssl': True,
+                       'with_winssl': False,
+                       'disable_threads': False,
+                       'with_ldap': False,
+                       'darwin_ssl': True,
+                       'with_libssh2': False,
+                       'with_libidn': False,
+                       'with_librtmp': False,
+                       'with_libmetalink': False,
+                       'with_libpsl': False,
+                       'with_largemaxwritesize': False,
+                       'with_nghttp2': False,
+                       'with_brotli': False
+                       }
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    _autotools = False
+
+    @property
+    def is_mingw(self):
+        return self.settings.os == "Windows" and self.settings.compiler != "Visual Studio"
+
+    def imports(self):
+        # Copy shared libraries for dependencies to fix DYLD_LIBRARY_PATH problems
+        #
+        # Configure script creates conftest that cannot execute without shared openssl binaries.
+        # Ways to solve the problem:
+        # 1. set *LD_LIBRARY_PATH (works with Linux with RunEnvironment
+        #     but does not work on OS X 10.11 with SIP)
+        # 2. copying dylib's to the build directory (fortunately works on OS X)
+
+        if self.settings.os == "Macos":
+            self.copy("*.dylib*", dst=self._source_subfolder, keep_path=False)
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+        # be careful with those flags:
+        # - with_openssl AND darwin_ssl uses darwin_ssl (to maintain recipe compatibilty)
+        # - with_openssl AND NOT darwin_ssl uses openssl
+        # - with_openssl AND with_winssl raises to error
+        # - with_openssl AND NOT with_winssl uses openssl
+        # Moreover darwin_ssl is set by default and with_winssl is not
+
+        if self.options.with_openssl:
+            # enforce shared linking due to openssl dependency
+            if self.settings.os != "Macos" or not self.options.darwin_ssl:
+                self.options["openssl"].shared = self.options.shared
+        if self.options.with_libssh2:
+            if self.settings.compiler != "Visual Studio":
+                self.options["libssh2"].shared = self.options.shared
+
+    def config_options(self):
+        if self.settings.os != "Macos":
+            self.options.remove("darwin_ssl")
+        if self.settings.os != "Windows":
+            self.options.remove("with_winssl")
+
+        if self.settings.os == "Windows" and self.options.with_winssl and self.options.with_openssl:
```

Comment:
You can't evaluate `self.options` in `config_options`. The design prevents only `self.settings` can be checked here.

Reply:
Moved to configure 👍